File: test_archives/py_tests/batch3/test_function4.py
from functions.function4 import function4
import pytest

def test_function4_1():
    assert function4(1, 2, 3.0, 4.0, 'hi', 'there', True, True) == '14hitherTrue'

def test_function4_2():
    assert function4(3, 4, 2.0, 4.1, 'hello', 'world', False, True) == '7hello worldFalse'

File: test_archives/cpp_tests/batch3/test_function4.cpp
#include "../catch.hpp"
#include "../functions/function4.cpp"

TEST_CASE("Test function4_1") {
    REQUIRE(function4(1, 2, 3.0, 4.0, "hi", "there", true, true) == "14hitherTrue");
}

TEST_CASE("Test function4_2") {
    REQUIRE(function4(3, 4, 2.0, 4.1, "hello", "world", false, true) == "7hello worldFalse");
}